concept/answer/AnswerGroup.java

```diff
@@ -19,17 +19,11 @@
 
 package grakn.client.concept.answer;
 
-import grakn.client.Grakn.Transaction;
 import grakn.client.concept.Concept;
-import grakn.client.concept.thing.impl.ThingImpl;
-import grakn.client.concept.type.impl.TypeImpl;
-import grakn.protocol.AnswerProto;
 
 import java.util.List;
 
-import static java.util.stream.Collectors.toList;
-
-public class AnswerGroup<T> implements Answer {
+public class AnswerGroup<T> {
```

Comment:
Should this not, now, be renamed to `ConceptMapGroup`, and a new class created named `NumberGroup`?

It's possible that they could inherit or implement `AnswerGroup<ConceptMap>` and `AnswerGroup<Number>`.

Reply:
I think in Java, we can be generic and therefore we should. We flatten the structure in protobuf because it could not be implemented generically there. I am keeping this implementation because this approach is consistent with what we have in Grakn Core: https://github.com/graknlabs/grakn/blob/master/query/QueryManager.java#L182-L204

What do you think of that?

Reply:
The thing is that what we have in Grakn Core, most likely, existed before our meeting yesterday. I'm thinking we would update both Core and the clients.

Reply:
Okay, I like that. Because we don't need something as generic as `AnswerGroup<T>` anyway.

Reply:
Addressed in https://github.com/graknlabs/client-java/pull/218/commits/16316dbec63658cb13c503319f3cf662ac6199d8